File: 01.SERVIDOR/ws_comercializadora_gr08/src/main/java/ec/edu/monster/service/AuthService.java
package ec.edu.monster.service;

import ec.edu.monster.dto.LoginRequest;
import ec.edu.monster.dto.LoginResponse;
import ec.edu.monster.model.Usuario;
import jakarta.enterprise.context.ApplicationScoped;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

@ApplicationScoped
public class AuthService {
    
    private static final Logger LOGGER = Logger.getLogger(AuthService.class.getName());
    
    private final Map<String, Usuario> usuarios;
    
    public AuthService() {
        usuarios = new HashMap<>();
        inicializarUsuarios();
    }
    
    private void inicializarUsuarios() {
        // Usuario MONSTER - Solo puede hacer operaciones de venta, no CRUD
        Usuario monster = new Usuario("MONSTER", "MONSTER9", "USER");
        usuarios.put("MONSTER", monster);
        
        // Usuario admin - Puede hacer CRUD completo
        Usuario admin = new Usuario("admin", "admin", "ADMIN");
        usuarios.put("admin", admin);
        
        LOGGER.info("Usuarios inicializados: MONSTER (USER), admin (ADMIN)");
    }
    
    public LoginResponse autenticar(LoginRequest request) {
        try {
            Usuario usuario = usuarios.get(request.getUsername());
            
            if (usuario == null) {
                LOGGER.warning("Usuario no encontrado: " + request.getUsername());
                return LoginResponse.fallido("Usuario no encontrado");
            }
            
            if (!usuario.isActivo()) {
                LOGGER.warning("Usuario inactivo: " + request.getUsername());
                return LoginResponse.fallido("Usuario inactivo");
            }
            
            if (!usuario.getPassword().equals(request.getPassword())) {
                LOGGER.warning("Contraseña incorrecta para usuario: " + request.getUsername());
                return LoginResponse.fallido("Contraseña incorrecta");
            }
            
            LOGGER.info("Autenticación exitosa para: " + request.getUsername() + " con rol: " + usuario.getRol());
            return LoginResponse.exitoso(usuario.getUsername(), usuario.getRol());
            
        } catch (Exception e) {
            LOGGER.severe("Error en autenticación: " + e.getMessage());
            return LoginResponse.fallido("Error en el proceso de autenticación");
        }
    }
    
    public boolean tieneRol(String username, String rolRequerido) {
        Usuario usuario = usuarios.get(username);
        return usuario != null && usuario.tieneRol(rolRequerido);
    }
    
    public boolean esAdmin(String username) {
        return tieneRol(username, "ADMIN");
    }
    
    public boolean esUsuarioValido(String username) {
        Usuario usuario = usuarios.get(username);
        return usuario != null && usuario.isActivo();
    }
}

File: SERVIDOR SOAP/ws_comer_dotnet_gr08/ec.edu.monster.service/AuthService.cs
using System.Collections.Generic;
using System.Linq;
using ec.edu.monster.model;
using ws_comer_dotnet_gr08;

namespace ec.edu.monster.service
{
    public class AuthService
    {
        private readonly Dictionary<string, Usuario> _usuarios;

        public AuthService()
        {
            _usuarios = new Dictionary<string, Usuario>();
            InicializarUsuarios();
        }

        private void InicializarUsuarios()
        {
            var monster = new Usuario("MONSTER", "MONSTER9", "USER");
            _usuarios["MONSTER"] = monster;

            var admin = new Usuario("admin", "admin", "ADMIN");
            _usuarios["admin"] = admin;
        }

        public LoginResponse Autenticar(LoginRequest request)
        {
            if (request == null || string.IsNullOrWhiteSpace(request.Username) || string.IsNullOrWhiteSpace(request.Password))
            {
                return LoginResponse.Fallido("Credenciales inválidas");
            }

            if (!_usuarios.TryGetValue(request.Username, out var usuario))
            {
                return LoginResponse.Fallido("Usuario no encontrado");
            }

            if (!usuario.Activo)
            {
                return LoginResponse.Fallido("Usuario inactivo");
            }

            if (usuario.Password != request.Password)
            {
                return LoginResponse.Fallido("Contraseña incorrecta");
            }

            return LoginResponse.Exitoso(usuario.Username, usuario.Rol);
        }

        public bool TieneRol(string username, string rolRequerido)
        {
            return _usuarios.TryGetValue(username, out var usuario) && usuario.TieneRol(rolRequerido);
        }

        public bool EsAdmin(string username)
        {
            return TieneRol(username, "ADMIN");
        }

        public bool EsUsuarioValido(string username)
        {
            return _usuarios.TryGetValue(username, out var usuario) && usuario.Activo;
        }

        public Dictionary<string, string> ListarUsuariosDescripcion()
        {
            return _usuarios.Values.ToDictionary(
                u => u.Username,
                u => u.Username == "MONSTER" ? "USER (Solo ventas, sin CRUD)" : "ADMIN (Acceso completo including CRUD)");
        }
    }
}
